landoapi/uplift.py

```diff
@@ -143,9 +160,25 @@ def get_uplift_conduit_state(
             f"Every uplifted patch must have an associated bug ID: {missing} do not."
         )
 
+    rev_ids = [revision["id"] for revision in stack_data.revisions.values()]
+    rev_ids_to_all_diffs = get_rev_ids_to_diffs(phab, rev_ids)
+
     stack = RevisionStack(set(stack_data.revisions.keys()), edges)
 
-    return stack_data, stack, target_repo
+    return stack_data, stack, target_repo, rev_ids_to_all_diffs
+
+
+def get_latest_non_commit_diff(diffs: list[dict]) -> dict:
+    """Given a list of diff dicts, return the latest diff that avoids bug 1865760."""
```

Comment:
This still references a bug number, it would be better to actually summarize the behaviour here instead. The context of the bug is nice to have as an explanation of why this method is implemented, but ideally the docstring describes what this method is doing, not just why.

I.e.
```python
"""Return a non-commit diff given a list of diff dicts."""
```

You can elaborate more in additional lines in the docstring.